src/content/docs/extensions/vector.mdx

```diff
@@ -0,0 +1,266 @@
+│ table name │ index name      │ index type │ property names    │ extension loaded │ index definition              │
+├────────────┼─────────────────┼────────────┼───────────────────┼──────────────────┼─────────────────────────────--┤
+│ Book       │ title_vec_index │ HNSW       │ [title_embedding] │ True             │ CALL CREATE_VECTOR_INDEX(...) │
+└────────────┴─────────────────┴────────────┴───────────────────┴──────────────────┴───────────────────────────────┘
+```
+
+## Advanced Usage
+
+### Filtered vector search
+Kuzu allows you to perform vector similarity search with filter predicates by combining the vector index
+with projected graphs.
+
+#### What is a projected graph?
+A projected graph is a **subgraph** (i.e., a subset of the original graph) that contains only the nodes and relationships that
+match the given table names and predicates.
+
+You can define a projected graph in Kuzu as follows:
+```cypher
+// Create a projected graph
+CALL CREATE_PROJECTED_GRAPH(
+    'projected_graph_name', // Name of the projected graph
+    {                      // Node tables to project
+        'table_name': {
+            'filter': 'predicate' // Optional predicate to filter nodes
+        }
+    },
+    [                      // Relationship tables to project
+        'table_name'
+    ]
+);
+// Drop a projected graph
+CALL DROP_PROJECTED_GRAPH('projected_graph_name');
+```
+
+**Predicate**
+
+There are several rules with projected graph predicates:
+- The predicate must depends only on its node/relationship table, i.e. predicates involving multiple tables is not supported.
+- Since we don't assign a variable to the node/relationship table, we use `n` to reference the node and `r` to reference the relationship table. So properties need to be in the form of `n.property_name` or `r.property_name`.
+
+**Life Cycle**
+
+A projected graph is kept alive until:
+- it is dropped explicitly; OR
+- the connection is closed (not persist on disk).
+
+**Lazy evaluation**
+
+We use lazy evaluation to evaluate the predicate. The predicate is only evaluated when the projected graph is queried. As
+a result, the project graph operation always runs instantly.
+
+#### Filtered vector search using a projected graph
+Here's an example that demonstrates how to perform vector search for only books published after 2010:
+
+```cypher
+// Step 1: Create a projected graph that filters books by publication year
+CALL CREATE_PROJECTED_GRAPH(
+    'filtered_book',   // Name of the projected graph
+    {'Book': {'filter': 'n.published_year > 2010'}},   // Projected node table Book with a filter on published_year. `n` is a place_holder here to reference the node table.
+    []   // No relationship tables can be projected.
+);
+
+// Step 2: Perform vector similarity search on the filtered subset
+In the `QUERY_VECTOR_INDEX` function, we can pass in the name of the projected graph as `table_name` parameter.
```

Comment:
What is the query vector created on? As in what is the user query? Is it still "quantum machine learning" like above? We should make that clear here, as all we see in the example below is the vector.